src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/PiglinListener.java

```diff
@@ -38,6 +48,30 @@ public void onEntityPickup(EntityPickupItemEvent e) {
         }
     }
 
+    @EventHandler
+    public void onPiglinDropItem(EntityDropItemEvent e) {
+        if (e.getEntity() instanceof Piglin) {
+            Piglin piglin = (Piglin) e.getEntity();
+            Set<ItemStack> drops = SlimefunPlugin.getRegistry().getBarterDrops(); 
+            
+            /*
+             * NOTE:  Getting a new random number each iteration because multiple items could have the same
+             * % chance to drop, and if one fails all items with that number will fail.  Getting a new random number
+             * will allow multiple items with the same % chance to drop.
+             */
+            
+            for (ItemStack is : drops) {
+                SlimefunItem sfi = SlimefunItem.getByItem(is);
+                if (sfi instanceof PiglinBarterDrop && ((PiglinBarterDrop)sfi).getBarteringLootChance() >= ThreadLocalRandom.current().nextInt(1,101)) {
```

Comment:
Why [1; 101] here?

Reply:
Per walshy
https://github.com/TheBusyBiscuit/Slimefun4/pull/2136#discussion_r463325850

Reply:
You can also just do greater than random(100). I think the [1; 100] here is a bit confusing, might even be helpful to add a comment there too

Reply:
done